pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -581,6 +583,19 @@ func nodeAddressForLB(node *v1.Node) (string, error) {
 	return addrs[0].Address, nil
 }
 
+//getStringFromServiceAnnotation searches a given v1.Service for a specific annotationKey and either returns the annotation's value or a specified defaultSetting
+func getStringFromServiceAnnotation(service *v1.Service, annotationKey string, defaultSetting string) string {
+	glog.V(4).Infof("getStringFromServiceAnnotation(%v, %v, %v)", service, annotationKey, defaultSetting)
+	if annotationValue, ok := service.Annotations[annotationKey]; ok {
```

Comment:
imo it is good that we do not check that. It allows creating lbaas resources without external floating ip. One use-case for instance is that we have other servers outside kubernetes in same openstack subnet and we want expose services there.

```
apiVersion: v1
kind: Service
metadata:
  labels:
    run: web-nofloating
  name: web-nofloating
  annotations:
    loadbalancer.openstack.org/floating-network-id: ""
  namespace: default
spec:
  selector:
    run: web
  ports:
  - port: 80
    name: https
    protocol: TCP
    targetPort: 80
  type: LoadBalancer
```

the result is:
```
% kubectl get svc -o wide
NAME             CLUSTER-IP       EXTERNAL-IP                   PORT(S)        AGE       SELECTOR
kubernetes       10.254.0.1       <none>                        443/TCP        3h        <none>
web-ext          10.254.45.205    192.168.1.86,193.xx.xxx.225   80:31277/TCP   3h        run=web
web-int          10.254.59.195    192.168.1.88,10.222.130.102   80:31338/TCP   3h        run=web
web-nofloating   10.254.107.195   192.168.1.89                  80:30913/TCP   27s       run=web
```